eds/plugins/github_provider.py

```diff
@@ -0,0 +1,143 @@
+import os
+from typing import List, Dict, Optional
+
+from github3 import login, enterprise_login
+from github3.github import GitHub, GitHubEnterprise
+from github3.orgs import Organization
+from github3.repos.repo import Repository
+from github3.repos.contents import Contents
+# from github3.repos.branch import Branch
+# from github3.git import CommitTree, Commit, Tree, Hash, Reference
+# from github3.repos.comparison import Comparison
+# from github3.users import User
+# from github3.exceptions import NotFoundError, ForbiddenError
+
+from eds.interfaces.vcs_provider import VcsProvider
+from eds.interfaces.plugin import Plugin
+
+
+class GithubProvider(VcsProvider):
+    """Github Provider implementation."""
+
+    def __init__(self, token_env_var: str = None, github_enterprise_url: str = None):
```

Comment:
all plugins will instantiate simply from a `yaml` Dict.

```
 def __init__(self, yaml: Dict):
```

the contents of `yaml` being the plugin's `properties` mapping from "eds.yml".

the schema for those properties needs to be specified in  `eds.interfaces.vcs_provider.VcsProvider.schema`

the work of specifying the schema is what  https://github.com/manheim/eds/issues/5 is for.  it might be easier to make an attempt at #5 first, then come back to this PR.

Reply:
Ok I changed this to use a yaml for the constuctor.